xbmc/music/MusicDatabase.cpp

```diff
@@ -13234,6 +13234,13 @@ int CMusicDatabase::GetOrderFilter(const std::string& type,
     }
   }
 
+  // Get the right tableview as if we are using strArtistSort the column name is ambiguous
+  std::string table{};
+  if (StringUtils::StartsWithNoCase(type, "album"))
+    table = "albumview.";
+  else if (StringUtils::StartsWithNoCase(type, "song"))
+    table = "songview.";
+
```

Comment:
Wouldn't the first token of name contain the view name you're looking for? or is it supposed to be a different table in this specific case?
If this must remain as is, then it's a detail but I would include only the table name with the trailing . and add the period to the string concatenation later.

Reply:
Even if it is, there would still be a bunch of `ifs` and `StringUtils::startsWith` to slice it up.  Using type just seems more straightforward and less resource intensive.  Checking `name` would be done for each entry in the `orderfields` in the loop which we really don't need to do.

Period needs adding whatever so I don't really see much difference in where it's done.